object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionTransfer.java

```diff
@@ -51,89 +54,41 @@ public ObjectConstructionTransfer(final CFAnalysis analysis) {
   @Override
   public TransferResult<CFValue, CFStore> visitMethodInvocation(
       final MethodInvocationNode node, final TransferInput<CFValue, CFStore> input) {
+
+    exceptionalStores = makeExceptionalStores(node, input);
     TransferResult<CFValue, CFStore> result = super.visitMethodInvocation(node, input);
+    Map<TypeMirror, CFStore> exceptionalStoresTmp = exceptionalStores;
```

Comment:
What is going on with this temp variable?

Reply:
`super.visitMethodInvocation` eventually leads to calls to `accumulate`, which is overridden to side-effect `exceptionalStores`. Because `makeExceptionalStores` needs a `TransferInput`, it has to be set here, not in `accumulate`, where it is used - `accumulate` only has access to the `TransferResult`.

I realized looking back at this that I could rewrite this to avoid the temporary variable. It should be a bit clearer now; I also added some documentation to the field that explains why it has to be `@Nullable`.